src/instance/minecraft/commands/SkillCommand.ts

```diff
@@ -35,7 +36,9 @@ export default {
       return `${context.username}, Invalid username! (given: ${givenUsername})`
     }
 
-    const parsedProfile = await getParsedProfile(context.clientInstance.app.hypixelApi, uuid)
+    const parsedProfile = await context.clientInstance.app.hypixelApi
+      .getSkyblockProfiles(uuid)
+      .then((res) => res.filter((p) => p.profileName)[0].me)
```

Comment:
This is a bug. revert the old function.
you are basically checking whether a profile has a name, which all profiles do... so it will return on the first profile.
What you need is to check for the SELECTED profile. the old function does that.